ivy/functional/ivy/linear_algebra.py

```diff
@@ -2426,7 +2436,54 @@ def tensordot(
         b: ivy.array(76.)
     }
     """
-    return current_backend(x1, x2).tensordot(x1, x2, axes=axes, out=out)
+    if batched_modes is not None:
+        axes = _get_valid_contraction_modes_for_axes(x1.shape, x2.shape, axes)
+        batched_modes = _get_valid_contraction_modes_for_batches(
+            x1.shape, x2.shape, batched_modes
+        )
+        return _tensordot_with_batched_modes(x1, x2, axes, batched_modes)
+    return current_backend(x1, x2).tensordot(
+        x1, x2, axes=axes, batched_modes=batched_modes, out=out
+    )
```

Comment:
This creates scalability problems e.g. if we decide to add a new backend in the future that does not have a native `tensordot` function then you would need to implement the function compositionally ( with and without `batch_modes`) for that backend. So, it's best we don't call the backend implementation here and implement the logic of `batch_modes` being None in ivy itself.

Reply:
actually all the backends have a batched_modes parameter but it's ignored so the non None batched_modes param only get handled in ivy without going to the backend

Reply:
What I meant was we might add a backend in the future that does not have a `tensordot` function in which case you would need to implement it compositionally in the backend. To avoid that, the compositional implementation should be able to handle both the cases (1) where `batch_modes` is `None` (2) `batch_modes` is not `None` and we shouldn't be deferring to the backend implementation using `current_backend(*args).tensordot(*args,**kwargs)`. Hope it makes sense now. Happy to discuss if you have counter thoughts as well :))

Reply:
I see, thanks for clarifying, so do you suggest, should I replace it with `ivy.tensordot(*args, **kwargs)` instead ?
and also I noticed that all other function in this file use `current_backend(*args)` in their implementation.

Reply:
Actually, we shouldn't be using either of those since they'll end up calling the same function again ( for a future backend that does not have a primary implementation as I mentioned earlier). Rather, we should be implementing the logic for `ivy.tensordot` when `batched_modes` is `None` in our ivy implementation. The idea is the the compositional implementation should be able to handle all possible inputs that can be passed to the function. Hope it makes sense now :)

Reply:
but what if the backend actually has a `tensordot` implementation what should we do then ?

Reply:
@hello-fri-end I modified `tensordot `as requested

Reply:
> but what if the backend actually has a `tensordot` implementation what should we do then ?

In that case, `ivy.tensordot` will actually be the backend implementation after the backend is set :) Again, refer to section about partial mixed functions in the deep-dive for more details :smile: